### DISCUSSION TITLE
Polaris schedule stopped working

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#963] [Nixie]
I've been running Nixie with two schedules (one for the pool & the other for the Polaris) for months  without any issues with the schedules.  Earlier this week, the Polaris schedule stopped working; i.e., the Polaris no longer turns on at the appointed hour.  I've tried deleting and re-creating & saving the schedule, doing a hard reboot of Nixie (shutdown command, breaker turned off, wait a bit, turn breaker back on), & soft reboot to no avail.  Below is the schedule itself, the Polaris circuit page, and a capture created this morning from about a minute prior to the scheduled turn on to about a minute after the scheduled turn on.  I can turn the Polaris on/off from the dashPanel or Home Assistant without any problems---just the schedule appears to be broken.

The Polaris does not turn off via the schedule if I start it manually, either.

I also noticed that following a hard reboot, the filter pump does not turn on automatically, although the "pool" light in dashPanel turns green.  Once again, I can turn the filter pump on & off via dashPanel and HA and it operates normally from that point on.  I don't know if the problems  are related or not.  The pool schedule works as expected, other than following a hard reboot.

Can you please help me diagnose & repair these issues?  Thank you for your help!

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/ee46e4c5-b549-4319-a2f1-47b16209e655)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/11dccc5f-e3f3-4053-bc43-6dd67ce2095d)

[replay (9)6.6.2024.zip](https://github.com/user-attachments/files/15637508/replay.9.6.6.2024.zip)

**Comment by rstrouse** [comment]
You currently have something trying to turn on and off circuit 7 on the socket interface.  This could be something you defined in REM.

**Reply by FlaMike** [reply]
If circuit 7 is also relay 7 on the relay board, it's  the drain line.  It's an actuated valve.  It has an egg timer of 10 minutes to prevent the pool from emptying if the system goes south.  There is also an automation in HA that opens and closes the drain line if a moisture sensor determines the pool is overflowing.  We have not had  any rain since the weekend, so I'm not sure what may be trying to close the valve.  Typically in HA if you try turning something  off that's  already off, it's  no harm no foul.

When I look at poolConfig.json, under Circuits, there is id 6 which is the Pool, id 2 is the Pool Light, id 3 is the Polaris, id 4 is the Acid Feeder (pump relay), and id5 is the Drain Line (actuated valve, mentioned above).  There is no id 6, 7, ... in the file.  How do I determine what circuit 7 may be?

The Nixie case fan is the only additional item defined in REM.  It toggles the fan relay when the CPU temperature exceeds 90 F.  This time of year, the fan is always on.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/3504286b-6f38-4e75-a2fe-4e9d4aa96474)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/b8cfffb6-acf7-4163-bf06-842b5f226257)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/f9de9139-5757-488b-826f-886d36ae2d45)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/d7ece3ba-3e10-4055-b4bc-eebce1e53216)

**Comment by rstrouse** [comment]
Check the feeds on your multi-relay board.

**Reply by FlaMike** [reply]
There are two feeds from the board:  4 is for the case fan & 7 is the drain line valve actuator (the bad boy circuit you noted).  I just changed it to send the value only when changed.  Could that have been the problem?

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/719188eb-0c46-43bd-9081-e639f0b8a9ac)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/7ba6606f-ecd9-4344-bd83-572463f4e197)

**Reply by rstrouse** [reply]
There is no circuit 7 the circuit id for the drain line is 5.

In the poolState.json I see the next run time set for the polaris schedule as follows.  At the time this was taken, the schedule should have been on.  Did the schedule show a green or a yellow indicator in dashPanel?

```
      "scheduleTime": {
        "calculated": true,
        "calculatedDate": "2024-06-06T00:00:00.000-0500",
        "startTime": "2024-06-06T08:30:00.000-0500",
        "endTime": "2024-06-06T09:00:59.999-0500",
        "shouldBeOn": true
      },

```

**Reply by FlaMike** [reply]
Oh, my!  I changed the circuit ID in the feed to 5.  Good catch.  I did not get a green or yellow dot in dashPanel when the schedule should have been running--just an empty dot.

**Reply by rstrouse** [reply]
The time was between 8:30am and 9:00am right and the pool was on since the polaris is defined as a master cleaner.  I assume there is a booster pump involved, is that the case?

**Reply by FlaMike** [reply]
Yes, you are correct on all points.  The filter pump runs from 8:00 AM to 8:00 PM, so there is no delay in running the booster pump for the Polaris at 8:30 AM even though it's a master cleaner.

**Reply by FlaMike** [reply]
This morning, the Polaris did not start again.  I had a capture going from about 8:15 until 8:45.  The Polaris schedule, as you noted is to run the cleaner from 8:30-9:00.  There was no green or yellow dot in either the Polaris schedule or the Polaris feature entry--just empty dots as before.

I manually turned on the Polaris prior to ending the capture.  Because the capture ran so long it is too large to attach.  So I put it on dropbox & am hoping you get get it from there.  Here's the link...  https://www.dropbox.com/scl/fi/34frmdbut49m1i3081xoq/replay-9-6.7.2024.zip?rlkey=819n96bcd4ssy35dbyypc6df8&st=rwj8u4hj&dl=0

BTW, it appears the capture from yesterday was also included in today's capture.  So you'll need to scroll through a bit to get to today's logs .  Sorry about that!

Thanks again for all of your help!

**Reply by rstrouse** [reply]
Pull njsPC.  I believe the issue was related to the order in which the schedules were being evaluated.  The Polaris schedule needed to be performing a delay calculation against the pool circuit because a master cleaner needs to evaluate the startup and solar delays.  Now all of the body circuit schedules are evaluated before any others.

**Reply by FlaMike** [reply]
OK.  Done!  

I also turned off the "cached" option for npm2 so the system will rebuild on restart.  (Yes, I did the entire npm2 startup dance to make sure it's using the correct version of ecosystem.config.js.)  I don't know if that was part of the problem--we had a *very* brief power interruption during a storm last weekend.  I don't know if the power fluctuation caused Nixie to crash and restart in a funky state or not (I'm guessing it may have).  In any event, since the system in my installation is reasonably stable and I'm not restarting or changing it frequently, I decided to have a bit of patience on the restarts & hopefully start with a clean system whenever Nixie goes down & restarts.

I did  a couple of "sudo reboot now" commands & everything came up as expected, including the filter pump restart, which also got funky this week.

In the past I had similar flaky behavior with the pool light schedule.  I was never able to determine the cause, so I just decided to use HA to control lighting.  Assuming the Polaris schedule operates correctly, I may go  back to using Nixie for light control.

Thanks for your time, help, & support!  Have a super weekend!!!

**Reply by FlaMike** [reply]
Still no love on the Polaris schedule.  Oddly, I added a light schedule last night & it ran just fine.  However, the Polaris did not turn on this morning.  I modified the schedule on/off time so I could capture it, running the cleaner for only 5 minutes.  The cleaner never turned on/off and the schedule "dot" in dashPanel never turned green.  Argh!  Not sure what the problem is with this schedule--I even deleted & re-added it.  Do I need to re-build njsPC after modifying the schedule?  I did not do so for the lights yesterday...

Anyway, the replay is attached & all help &  suggestions are welcome!  I can always use HA to run the Polaris, but I'm totally flummoxed (not generally hard to do).

Thanks!

[6.9.2024Replay2.zip](https://github.com/user-attachments/files/15753049/6.9.2024Replay2.zip)

**Reply by FlaMike** [reply]
OK, I gave up on trying to repair the existing configuration as it was heading nowhere towards reliability quickly.  Schedules failed to start for both lights & the Polaris for no reason I was able to determine.  Oddly, the overall "pool" schedule (filter pump turn on & turn off) never failed.  But everything else did.

Fortunately, I had made a back micro SD card several weeks ago.  I installed the backup card, corrected the circuit for the drain line, as suggested by @rstrouse several posts up, added/updated the schedules for the lights & Polaris and...  Everything has worked as expected for the past two days!  

I don't know what happened to the previously working configuration/SD card/whatever, but the backup, properly configured seems to have done the trick.  I suspect a brief power fluctuation due to a thunderstorm was the source of the problem, but ???

Anyway, the lesson learned for me at least, is  _always keep a  backup of your current installation and configuration available_.  Fortunately, I had one that was very close to "current."  Still, I took screenshots of every configuration item in dashPanel and REM and made certain the newly installed SD card was updated to match my most recent setup.

I've already made a backup of the working currently running SD card and am actually using it to run the pool as I write this note (secondary lesson, _always confirm that your backup works_!)

For those of you wondering the easiest & quickest way to clone your SD card, I suggest using Balena Etcher.  It's free & available for Linux, Mac, & Windows.  Put your source SD card in your PC, click "clone disk" in Balena Etcher, select the SD card you just put in the PC as the source, click select destination, then insert the target SD card & select it, click Flash/Start/Clone (whatever the button says) & wait a couple of minutes.  When it's done, you have two copies of your working setup.  Put one back in your Pi & fire it up.  Mark the other SD card as your backup & put it in a safe place.  It may save you a lot of grief & hair pulling.

That's my story & I'm sticking to it!
